### ISSUE TITLE
JSON Compare is failing even when using `CompareOrder.Lenient`

### ISSUE THREAD

**Issue by busches** [kotest/kotest#2626]
**Which version of Kotest are you using**
4.6.3

I have a basic example of trying to compare a nested array of json objects and it's failing. As you can see `myIds` have the same values, but the order is flipped.

Example:
```kotlin
val expected = """
{
  "pageSize": 50,
  "page": 1,
  "totalCount": 2,
  "myIds": [
    {
      "id": "4b167e61-d3b1-4f80-b5ec-fcec65cf5d95"
    },
    {
      "id": "f0f2d5ad-5a56-46e1-bbdf-06b58ec3a350"
    }
  ]
}
""".trimIndent()
val actual = """
{
  "myIds": [
    {
      "id": "f0f2d5ad-5a56-46e1-bbdf-06b58ec3a350"
    },
    {
      "id": "4b167e61-d3b1-4f80-b5ec-fcec65cf5d95"
    }
  ],
  "pageSize": 50,
  "page": 1,
  "totalCount": 2
}
""".trimIndent()

actual.shouldEqualJson(expected, CompareMode.Lenient, CompareOrder.Lenient)
```

Error:
```
At 'myIds.[0].id' expected '4b167e61-d3b1-4f80-b5ec-fcec65cf5d95' but was 'f0f2d5ad-5a56-46e1-bbdf-06b58ec3a350'
java.lang.AssertionError: At 'myIds.[0].id' expected '4b167e61-d3b1-4f80-b5ec-fcec65cf5d95' but was 'f0f2d5ad-5a56-46e1-bbdf-06b58ec3a350'
```

According to the [docs](https://kotest.io/docs/assertions/json-matchers.html) I likely want to use `CompareOrder.LenientAll` but this does not exist in 4.6.3 from what I can tell, it was added with https://github.com/kotest/kotest/pull/2518.

Is this another case of the docs are showing 5.x features or is this a bug? If `CompareOrder.LenientAll` doesn't exist in `4.6.3`, is there a way to get the above example to work?

Even using: `JSONObject(actual).get("myIds").toString().shouldEqualJson(JSONObject(expected).get("myIds").toString(), CompareMode.Lenient, CompareOrder.Lenient)` still fails.

**Comment by Kantis** [comment]
> Is this another case of the docs are showing 5.x features or is this a bug? If CompareOrder.LenientAll doesn't exist in 4.6.3, is there a way to get the above example to work?

Yes, sorry for the inconvenience. I'll revert the docs and add to the 5.0 docs PR

I don't think there's any built-in way of handling out-of-order arrays in `4.6.x`.

**Comment by sksamuel** [comment]
@Kantis is this all ok with 5.0 and just a docs issue ?
Ie can this be closed ?

**Comment by busches** [comment]
It's just docs. With 1.6 out today, hopefully 5.0 is around the corner.

**Comment by sksamuel** [comment]
Yep, just upgraded Kotest to 1.6
Going to set 5.0.0.RC going shortly.

**Comment by Kantis** [comment]
Yeah, just docs. The issue is fixed with 5.0. Left it open in case we would make another 4.x release
